stingray/simulator/base.py

```diff
@@ -1,14 +1,24 @@
-
 import numpy as np
 import numpy.random as ra
 import scipy.interpolate as sci
-from ..utils import assign_value_if_none
+from stingray import Lightcurve
+from stingray.gti import gti_border_bins
 
 
-def simulate_times(lc, use_spline=False, bin_time=None):
+def simulate_times(lc, use_spline=False):
     """
-    Assign (simulate) photon arrival times to event list, using the
-    acceptance-rejection method.
+    Simulate an event list, by using the inverse CDF method:
+
+    + Assume that the light curve is a probability density (must be positive
+      definite)
+
+    + Calculate the CDF from the cumulative sum, and normalize it from 0 to 1
+
+    + Extract N random probability values from 0 to 1
+
+    + Find the CDF values corresponding to these N values
+
+    + Find the times corresponding to these N CDF values
```

Comment:
I am curious how this compares performance-wise to the previous method. Have you done any performance tests?

Reply:
Yes, there is a huge improvement and, more importantly, this method is much more predictable. You need to only simulate the number of events you need, while in the other case you needed many more (because some of them got accepted, some rejected, and if the lc had very strong variability the ones rejected were many!)